### PR TITLE
another fix for the calendar

### PR DESCRIPTION
**Pull request by voidus**

### CODE REVIEW

**Comment by voidus** [comment]
The problem is that using sort() here is extremely inefficient (almost near n²) because we have sorted sub-sequences.
I'll patch this up a bit nicer, though. It was late when I wrote this.

**Comment by voidus** [comment]
I think I made it a bit clearer now. It should also perform better.

File: phprojekt/application/Calendar2/Helper/Rrule.php [hunk 1]
```diff
@@ -195,6 +188,55 @@ public function getDatesInPeriod(Datetime $start, Datetime $end)
...
+            $datestring = $date->format('Y-m-d H:i:s');
+            $date       = new Datetime($datestring, new DateTimeZone('UTC'));
+
+            $ts = $date->getTimestamp();
+            if ($startTs <= $ts + $this->_duration
+                    && $ts <= $endTs
+                    && !in_array($date, $this->_exceptions)) {
+                $dateSeries[$k][] = $ts;
+            } else if ($ts > $endTs) {
+               break;
```
Comment (exi) [inline phprojekt/application/Calendar2/Helper/Rrule.php:210]: i think the break is not indented correctly

File: phprojekt/application/Calendar2/Helper/Rrule.php [hunk 2]
```diff
@@ -195,6 +188,55 @@ public function getDatesInPeriod(Datetime $start, Datetime $end)
...
+     *  - All arrays in $sequences have the same spacing between events,
+     *      i.e. the interval used to create the dateperiods is the same.
+     *  - $sequences is sorted in regard to the first element of each inner array
+     */
+    private function _mergeSequences(array $sequences)
+    {
+        $ret = array();
+
+        foreach (array_keys($sequences) as $k) {
+            $ret[] = array_shift($sequences[$k]);
```
Comment (exi) [inline phprojekt/application/Calendar2/Helper/Rrule.php]: Do i understand this correctly?
You don't merge all elements from $sequences[$k] but only the first element of every sequence?
This is not very clear in the comments

**Pull request merged**